### PR TITLE
Document generated struct constructors

### PR DESCRIPTION
**Pull request by DanielKamkha**
Signed-off-by: Daniel Kamkha <daniel.kamkha@here.com>

### CODE REVIEW

File: docs/struct_constructors.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
+Struct constructors
+===================
+
+This documents various types of constructors that can be defined for structs in LIME IDL.
+
+Implicit constructors
```
Comment (Hsilgos) [inline docs/struct_constructors.md:6]: I would add that implicit and field constructors are generated in specific language only without crossing languages boundaries.
Also despite that it's quite obvious for implicit constructors I would specify order in which parameters appear in constructor

File: docs/struct_constructors.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    isMultiline: Boolean = true
+}
+```
+will have both `MyStruct(text)` and `MyStruct(text, isMultiline)` constructors generated.
+
+Implicit constructors are sufficient for most use cases. When the API semantics require a different set of constructors,
+or when constructors need to have logic beyond just field initialization, explicit constructor definitions could be used
+instead, as outlined in the following sections.
+
+Field constructors
```
Comment (Hsilgos) [inline docs/struct_constructors.md:34]: Is definition of field and/or custom constructor preserves implicit ones?

File: docs/struct_constructors.md (hunk 1 above)
Comment (DanielKamkha) [reply docs/struct_constructors.md:6]: Updated

File: docs/struct_constructors.md (hunk 2 above)
Comment (DanielKamkha) [reply docs/struct_constructors.md:43]: Updated

**Review by Hsilgos — APPROVED** [review]

**Pull request merged**